src/auth/googleauth.ts

```diff
@@ -451,6 +451,13 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
       );
     }
 
+    // No ADC, but perhaps there's an API key available...
+    if ('apiKey' in options && options.apiKey) {
```

Comment:
I actually think API Key should override ADC:

```Additionally, if application default credentials (ADC) are enabled and an API key is specified, ADC should be ignored.```

Also, we should add an error situation if an API Key and credentials are used together:

```API Keys and Credentials are mutually exclusive authentication methods and cannot be used together.```